Sources/Controllers/Resources/OAManageResourcesViewController.mm

```diff
@@ -996,24 +1037,24 @@ - (OAResourceItem *) collectWorldSeamarksItem
     return nil;
 }
 
-- (BOOL)isMapResource:(const OsmAnd::ResourcesManager::ResourceType)resourceType
-{
-    return resourceType == OsmAndResourceType::MapRegion ||
-        resourceType == OsmAndResourceType::RoadMapRegion ||
-        resourceType == OsmAndResourceType::SrtmMapRegion ||
-        resourceType == OsmAndResourceType::WikiMapRegion ||
-        resourceType == OsmAndResourceType::SlopeRegion ||
-        resourceType == OsmAndResourceType::HillshadeRegion;
-}
-
 - (void) collectResourcesDataAndItems
 {
     [self collectSubregionItems:self.region];
     
     [_allResourceItems addObjectsFromArray:_allSubregionItems];
     [_allResourceItems sortUsingComparator:self.resourceItemsComparator];
-    [_regionMapItems sortUsingComparator:self.resourceItemsComparator];
-    
+
+    _regionMapItems = [[_regionMapItems sortedArrayUsingComparator:^NSComparisonResult(OAResourceItem *res1, OAResourceItem *res2) {
+        NSInteger orderValue1 = [OAResourceType getOrderIndex:[OAResourceType toValue:res1.resourceType]];
+        NSInteger orderValue2 = [OAResourceType getOrderIndex:[OAResourceType toValue:res2.resourceType]];
+        if (orderValue1 < orderValue2)
+            return NSOrderedAscending;
+        else if (orderValue1 > orderValue2)
+            return NSOrderedDescending;
+        else
+            return NSOrderedSame;
+    }] mutableCopy];
```

Comment:
Mutable arrays can be sorted in place without the need of copying anything (see here and in other places throughout the code)